rosbag2/src/rosbag2/writer.cpp

```diff
@@ -59,7 +71,10 @@ void Writer::open(
   if (!storage_) {
     throw std::runtime_error("No storage could be initialized. Abort");
   }
+
   uri_ = storage_options.uri;
+  start_time_ = std::chrono::nanoseconds::max().count();
+  end_time_ = std::chrono::nanoseconds::min().count();
```

Comment:
This could directly be set in a metadata struct.

Reply:
I pulled it out because I thought less characters per line would be more clean. After looking at it again, I think I should move both lines inside `if (storage_)` block.